src/main/telemetry/srxl.c

```diff
@@ -160,6 +162,168 @@ bool srxlFrameRpm(sbuf_t *dst, timeUs_t currentTimeUs)
+}
+
+/*
+typedef struct
+{
+    UINT8    identifier;    // Source device = 0x16
+    UINT8    sID;           // Secondary ID
+    UINT16   altitudeLow;   // BCD, meters, format 3.1 (Low order of altitude)
+    UINT32   latitude;      // BCD, format 4.4, Degrees * 100 + minutes, less than 100 degrees
+    UINT32   longitude;     // BCD, format 4.4 , Degrees * 100 + minutes, flag indicates > 99 degrees
+    UINT16   course;        // BCD, 3.1
+    UINT8    HDOP;          // BCD, format 1.1
+    UINT8    GPSflags;      // see definitions below
+} STRU_TELE_GPS_LOC;
+
+// GPS flags definitions:
+#define GPS_INFO_FLAGS_IS_NORTH_BIT                 (0)
+#define GPS_INFO_FLAGS_IS_EAST_BIT                  (1)
+#define GPS_INFO_FLAGS_LONGITUDE_GREATER_99_BIT     (2)
+#define GPS_INFO_FLAGS_GPS_FIX_VALID_BIT            (3)
+#define GPS_INFO_FLAGS_GPS_DATA_RECEIVED_BIT        (4)
+#define GPS_INFO_FLAGS_3D_FIX_BIT                   (5)
+#define GPS_INFO_FLAGS_NEGATIVE_ALT_BIT             (7)
+*/
+
+bool srxlFrameGpsLoc(sbuf_t *dst, timeUs_t currentTimeUs)
+{
+    gpsCoordinateDDDMMmmmm_t coordinate;
+    uint32_t latitudeBcd, longitudeBcd, altitudeLo;
+    uint16_t altitudeLoBcd, groundCourseBcd, hdop;
+    uint8_t x, hdopBcd;
+    uint8_t gpsFlags = 0x00;
+
+    UNUSED(currentTimeUs);
+
+    // lattitude
+    GPStoDDDMM_MMMM(gpsSol.llh.lat, &coordinate);
+    latitudeBcd  = (dec2bcd_r(coordinate.dddmm) << 16) | dec2bcd_r(coordinate.mmmm);
+
+    // longitude
+    GPStoDDDMM_MMMM(gpsSol.llh.lon, &coordinate);
+    longitudeBcd = (dec2bcd_r(coordinate.dddmm) << 16) | dec2bcd_r(coordinate.mmmm);
+
+    // altitude
+    altitudeLo = ABS(gpsSol.llh.altCm) / 10;
+    if (altitudeLo > 999999) altitudeLo = 999999;
+    x = altitudeLo / 100000;
+    if (x > 0) altitudeLo-=(x * 100000);
+    x = altitudeLo / 10000;
+    if (x > 0) altitudeLo-=(x * 10000);
+    altitudeLoBcd = dec2bcd_r(altitudeLo);
+
+    // Ground course
+    groundCourseBcd = dec2bcd_r(gpsSol.groundCourse);
+
+    // HDOP
+    hdop = gpsSol.hdop / 10;
+    if (hdop > 99) hdop = 99;
+    hdopBcd = (dec2bcd_r(hdop));
+
+    // flags
+    if (gpsSol.llh.lat > 0) gpsFlags = gpsFlags | 0x01;   // North
+    if (gpsSol.llh.lon > 0) gpsFlags = gpsFlags | 0x02;   // East
+    if (STATE(GPS_FIX)) gpsFlags = gpsFlags     | 0x28;   // GPS Fix
+    if (gpsSol.llh.altCm < 0) gpsFlags = gpsFlags | 0x80; // Negative altitude
+    if ((gpsSol.llh.lon / GPS_DEGREES_DIVIDER) > 99) 
+        gpsFlags = gpsFlags                     | 0x04;   // Longitude > 99
+    gpsFlags = gpsFlags                         | 0x10;   // Data received bit
+
+    // SRXL frame
+    sbufWriteU8(dst, SRXL_FRAMETYPE_GPS_LOC);
+    sbufWriteU8(dst, SRXL_FRAMETYPE_SID);
+    sbufWriteU16(dst, altitudeLoBcd);
+    sbufWriteU32(dst, latitudeBcd);
+    sbufWriteU32(dst, longitudeBcd);
+    sbufWriteU16(dst, groundCourseBcd);
+    sbufWriteU8(dst, hdopBcd);
+    sbufWriteU8(dst, gpsFlags);
+
+    return true;
```

Comment:
This will cause GPS tm frames to be sent unconditionally even if there are no GPS data available and even if no GPS unit is connected at all. You should check for available data and only then `return true` else `return false` to avoid wasting precious tm bandwidth. Compare how `srlFrameVTX()` and  `srxlFrameFlightPackCurrent()` is done.